wordle-app/pages/WordEditor.vue

```diff
@@ -0,0 +1,245 @@
+<template>
+  <v-container style="max-width: 800px">
+    <v-toolbar flat class="mb-3">
+      <v-toolbar-title>Word Editor</v-toolbar-title>
+      <v-spacer></v-spacer>
+      <v-switch
+        label="Display common words only"
+        color="secondary"
+        @change="fetchWords"
+        class="d-flex justify-center"
+        v-model="isCommonFilter"
+      />
+      <v-btn
+        v-if="isLogged"
+        color="primary"
+        prepend-icon="mdi-plus-box"
+        variant="flat"
+        class="ml-5"
+        @click="showDialog = true"
+        >Add Word
+      </v-btn>
+    </v-toolbar>
+
+    <v-data-table-server
+      v-model:items-per-page="options.itemsPerPage"
+      :headers="headers"
+      :items="words"
+      :items-length="totalWords"
+      :options.sync="options"
+      @update:options="fetchWords"
+      @update:page="pageNumber"
+      @update:items-per-page="itemsPerPage"
+      :loading="isLoading"
+      :items-per-page-options="[10, 25, 50, 100]"
+      hover
+      density="compact"
+    >
+      <template #top>
+        <v-text-field
+          v-model="search"
+          label="Search"
+          @input="fetchWords"
+          clearable
+          clear-icon="mdi-close-circle"
+          variant="solo-filled"
+          bg-color="white"
+          prepend-inner-icon="mdi-magnify"
+        />
+      </template>
+
+      <!-- Template for Word text -->
+      <template #item.text="{ item }">
+        <p class="font-weight-bold">
+          {{ item.text.charAt(0).toUpperCase() + item.text.slice(1) }}
+        </p>
+      </template>
+
+      <template #item.actions="{ item }">
+        <v-btn @click="deleteItem(item)" color="red">Delete</v-btn>
+      </template>
+
+      <!-- Template for switch to change isCommon -->
+      <template #item.isCommon="{ item }">
+        <v-switch
+          class="d-flex justify-center"
+          v-model="item.isCommon"
+          color="green"
+          inset
+          @change="updateItem(item)"
+          v-if="isLogged"
+        />
+
+        <v-icon v-else :color="item.isCommon ? 'green' : 'red'" class="my-4">
+          {{ item.isCommon ? "mdi-check" : "mdi-close" }}
+        </v-icon>
+      </template>
+    </v-data-table-server>
+
+    <AddWordDialog @save="saveWord" v-model:show="showDialog" />
+  </v-container>
+</template>
+
+<script setup lang="ts">
+import Axios from "axios";
+import TokenService from "~/scripts/tokenService";
+
+// Log in stuff here
+const tokenService = new TokenService();
+const isLogged = ref(tokenService.isLoggedIn());
+
```

Comment:
The only thing that is throwing me off is the word editor list is alphabetized but it seems the alphabet starts at F and not at A, after searching for ABCDE it doesn't seem to retrieve any of the words with those starting letters, which leads me to believe your database/wordlist is being cut off or they got removed somewhere!

Reply:
This is weird. On my local machine, this feature works as expected. I also noticed that there was only half of the initial word list in my Azure DB. It feels like someone deletes them. But I will take another look at this feature. Thank you for the great review.